js/src/dapps/registry/Events/events.js

```diff
@@ -42,57 +46,114 @@ const renderStatus = (timestamp, isPending) => {
+      </td>
+      <td>
+        <code>
+          <Hash hash={ bytesToHex(e.parameters.name.value) } />
+        </code>
+      </td>
+      <td>
+        { renderStatus(e.timestamp, e.state === 'pending') }
+      </td>
     </tr>
   );
 };
 
-const renderDataChanged = (e, accounts, contacts) => {
+const renderDataChanged = (e) => {
   let classNames = styles.dataChanged;
   if (e.state === 'pending') {
     classNames += ' ' + styles.pending;
   }
 
   return (
     <tr key={ e.key } className={ classNames }>
-      <td>{ renderAddress(e.parameters.owner.value, accounts, contacts) }</td>
-      <td><abbr title={ e.transaction }>updated</abbr></td>
       <td>
-        key <code>{ new Buffer(e.parameters.plainKey.value).toString('utf8') }</code> of <code>{ renderHash(bytesToHex(e.parameters.name.value)) }</code>
+        <Address address={ e.parameters.owner.value } />
+      </td>
+      <td>
+        <abbr title={ e.transaction }>updated</abbr>
+      </td>
+      <td>
+        { 'key ' }
+        <code>
+          { new Buffer(e.parameters.plainKey.value).toString('utf8') }
+        </code>
+        { 'of ' }
+        <code>
+          <Hash hash={ bytesToHex(e.parameters.name.value) } />
+        </code>
+      </td>
+      <td>
+        { renderStatus(e.timestamp, e.state === 'pending') }
+      </td>
+    </tr>
+  );
+};
+
+const renderReverse = (e) => {
+  const verb = ({
+    ReverseProposed: 'proposed',
+    ReverseConfirmed: 'confirmed',
+    ReverseRemoved: 'removed'
+  })[e.type];
+  if (!verb) return null;
+
+  const classNames = styles.reverse + (e.state === 'pending' ? ` ${styles.pending}` : '');
+  const details = [
+    'name ',
+    // TODO: this is an indexed param, cannot display as plain text
+    (<code key='name'>{ bytesToHex(e.parameters.name.value) }</code>),
+    ' for ',
+    (<Address key='reverse' address={ e.parameters.reverse.value } />)
+  ];
+
+  return (
+    <tr key={ e.key } className={ classNames }>
+      <td>
+        <Address address={ e.from } />
+      </td>
+      <td>{ verb }</td>
+      <td>{ details }</td>
+      <td>
+        { renderStatus(e.timestamp, e.state === 'pending') }
       </td>
-      <td>{ renderStatus(e.timestamp, e.state === 'pending') }</td>
     </tr>
   );
 };
 
 const eventTypes = {
   Reserved: renderEvent(styles.reserved, 'reserved'),
   Dropped: renderEvent(styles.dropped, 'dropped'),
-  DataChanged: renderDataChanged
+  DataChanged: renderDataChanged,
+  ReverseProposed: renderReverse,
+  ReverseConfirmed: renderReverse,
+  ReverseRemoved: renderReverse
 };
 
-export default class Events extends Component {
+class Events extends Component {
 
   static propTypes = {
-    actions: PropTypes.object.isRequired,
     subscriptions: PropTypes.object.isRequired,
     pending: PropTypes.object.isRequired,
     events: PropTypes.array.isRequired,
-    accounts: PropTypes.object.isRequired,
-    contacts: PropTypes.object.isRequired
+
```

Comment:
It's not random. The distinction between props (=state) and actions (=state mutating things) is pretty clear, throughout the whole UI code.